tools/tnodb_mock/provision.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/threefoldtech/zosv2/modules/provision"
+)
+
+type reservation struct {
+	Reservation *provision.Reservation
+	Sent        bool
+}
+
+var reservationStore map[string][]*reservation
+var reservationMu sync.Mutex
+
+func reserve(w http.ResponseWriter, r *http.Request) {
+	nodeID := mux.Vars(r)["node_id"]
+
+	_, ok := nodeStore[nodeID]
+	if !ok {
+		http.Error(w, fmt.Sprintf("node %s not found", nodeID), http.StatusNotFound)
+		return
+	}
+
+	defer r.Body.Close()
+	res := &provision.Reservation{}
+	if err := json.NewDecoder(r.Body).Decode(res); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	reservationMu.Lock()
+	defer reservationMu.Unlock()
+	reservationStore[nodeID] = append(reservationStore[nodeID], &reservation{Reservation: res})
+	w.WriteHeader(http.StatusCreated)
+}
+
+func getReservations(w http.ResponseWriter, r *http.Request) {
+	nodeID := mux.Vars(r)["node_id"]
+
+	_, ok := nodeStore[nodeID]
+	if !ok {
+		http.Error(w, fmt.Sprintf("node %s not found", nodeID), http.StatusNotModified)
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	ch := make(chan []*provision.Reservation)
+
+	// start long polling
+	go func(ctx context.Context, ch chan<- []*provision.Reservation) {
```

Comment:
I don't get why this has to be in a go routine, it can be just a busy loop with intermittent sleeps

Reply:
Indeed, each request is already executing in its own goroutine. thx